tests/release/e2e-test-push-image-to-pyxis.go

```diff
@@ -258,6 +258,8 @@ var _ = framework.ReleaseSuiteDescribe("[HACBS-1571]test-release-e2e-push-image-
 					return false
 				}
 
+				Expect(utils.PipelineRunFailed(releasePr)).NotTo(BeTrue(), fmt.Sprintf("did not expect PipelineRun %s:%s to fail", releasePr.GetNamespace(), releasePr.GetName()))
```

Comment:
I have another idea. How about we just remove the `GetCondition(apis.ConditionSucceeded).IsTrue()` from the `return`? We would just wait for the prs to finish here. And then the next step would be to verify that they succeeded.

I would say it would be a more straightforward workflow. On the other hand, the downside would be that we would always wait for both PRs to finish first, so it might take slightly longer.

WDYT?

Reply:
hmm I'm not sure about the value of having a spec for validating only that both PipelineRuns have finished.

If one of them finishes earlier and has condition "Failed", it doesn't make sense to proceed with the test (i.e. wait for the other PipelineRun), so IMO the test should fail immediately.

Reply:
I think that Martin meant to remove  `GetCondition(apis.ConditionSucceeded).IsTrue()` from the return like below 
				```
return releasePr.HasStarted() && releasePr.IsDone() &&
					additionalReleasePr.HasStarted() && additionalReleasePr.IsDone() 
```
and this cause the exception we added of any pr would fail, am I correct @mmalina 

It does make sense Martin's suggestion just in the current way we ensure that the PRs or on **failure** pr **Success**   so we are on the safe side.

Reply:
Yes, in my opinion the condition in the successful PR is not right. In fact, I think we would be better off doing it one by one:
```
repeat:
    get the first pipelinerun's status
until it has started and is finished

assert/expect the first pr to succeed

repeat:
    get the second pipelinerun's status
until it has started and is finished

assert/expect the second pr to succeed
```

It's a bit more verbose, but I think it would be clearer what is going on, no?

Reply:
My suggestion was about making sure that the test spec doesn't fail later than it has to (in case that the PR2 would take longer, which could be usually a matter of seconds)

Any solution that would implement the logic that the spec fails immediately if any of the pipelineruns failed is fine by me though. It'd be definitely better than what we had here before.

What Martin is suggesting is currently implemented in the [has controller](https://github.com/redhat-appstudio/e2e-tests/blob/bc8f0d54149a7de8172e6fda2d47b24abba14786/pkg/utils/has/controller.go#L503-L516) (with the difference that `wait.PollImmediate` is used there instead of `Eventually` wrapper, that we use directly in tests). IMO it would make sense to implement such method for Release controller, or perhaps reuse what we have in [tekton controller](https://github.com/redhat-appstudio/e2e-tests/blob/bc8f0d54149a7de8172e6fda2d47b24abba14786/pkg/utils/tekton/controller.go#L238) (since we know the PipelineRun name)

@kasemAlem do you plan to implement this in other release tests as well?

Reply:
OK. For this PR I'm fine with the original proposal of failing immediately inside the `Eventually` loop.

Reply:
thanks both for the great comments, 
@psturc  sounds a great way to implement in release